common/cah-model.cpp

```diff
@@ -0,0 +1,183 @@
+                    device_calibration dev_cal(dev_model.dev);
+                    // Register AC status change callback
+                    if (!registered_to_callback)
+                    {
+                        registered_to_callback = true;
+                        cah_state = model_state_type::PROCESS_MODAL;
+                        // We switch to process state without a guarantee that the process really started,
+                        // Set a timeout to make sure if it is not started we will allow closing the window.
+                        cah_process_timeout.start(std::chrono::seconds(30)); 
+                        dev_cal.register_calibration_change_callback([&](rs2_calibration_status cal_status)
+                        {
+                            calib_status = cal_status;
+                        });
+                    }
+                }
+            }
+            else
+            {
+                keep_showing = false;
+            }
+        }
+    }
+    break;
+
+    case model_state_type::PROCESS_MODAL:
+    {
+        if (!process_started)
+        {
+            // Indication of calibration process start
+            process_started = (calib_status == RS2_CALIBRATION_SPECIAL_FRAME || calib_status == RS2_CALIBRATION_STARTED);
+        }
+
+        bool process_finished(calib_status == RS2_CALIBRATION_SUCCESSFUL || calib_status == RS2_CALIBRATION_FAILED || calib_status == RS2_CALIBRATION_NOT_NEEDED);
+
+        static std::map<rs2_calibration_status, std::string> status_map{
+            {RS2_CALIBRATION_SPECIAL_FRAME  , "In Progress" },
+            {RS2_CALIBRATION_STARTED        , "In Progress" },
+            {RS2_CALIBRATION_NOT_NEEDED     , "Ended" },
+            {RS2_CALIBRATION_SUCCESSFUL     , "Ended Successfully" },
+            {RS2_CALIBRATION_RETRY          , "In Progress" },
+            {RS2_CALIBRATION_FAILED         , "Ended With Failure" },
+            {RS2_CALIBRATION_SCENE_INVALID  , "In Progress" },
+            {RS2_CALIBRATION_BAD_RESULT     , "In Progress" } };
+
+        rs2_calibration_status calibration_status = calib_status;
+
+        // We don't know if AC really started working so we add a timeout for not blocking the viewer forever.
+        if (!process_started)
+        {
+            if (cah_process_timeout.has_expired())
+            {
+                process_finished = true; // on timeout display failure and allow closing the window
+                calibration_status = RS2_CALIBRATION_FAILED;
+            }
+        }
+
+        const std::string & message = process_finished ? "                                                               " :
+                                                         "Camera Accuracy Health is In progress, this may take a while...";
+        if (status_dialog("Camera Accuracy Health Status", message, status_map[calibration_status], process_finished, window))
+        {
+            keep_showing = false;
+        }
+    }
+    break;
```

Comment:
alignment issue?

Reply:
No.. the break is in place, after the curly bracket .